src/App.tsx

```diff
@@ -2,31 +2,59 @@ import React from "react";
 import logo from "./logo.svg";
 import "./App.scss";
 import "./tailwind.css";
+import { fireAuth, fireAuthSignOut } from "./FireConfig";
 
 import Onboard from "./components/Onboard";
+import Dashboard from "./components/Dashboard";
 
-import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
+import { BrowserRouter as Router, Switch, Route, Link, useHistory } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import ChangePassword from "components/ChangePassword";
+import RegisterUsers from "components/RegisterUsers";
+
+import authSlice from 'store/FireActions';
+import store from './store/store';
 
 function App() {
+  const currentUser = useSelector((state: any) => state.auth.userUID);
```

Comment:
State should be typed. See [here](https://redux-toolkit.js.org/usage/usage-with-typescript) for more information.